tapgarden/planter_test.go

```diff
@@ -309,7 +309,12 @@ func (t *mintingTestHarness) queueSeedlingsInBatch(isFunded bool,
 		// The received update should be a state of MintingStateSeed.
 		require.Equal(t, tapgarden.MintingStateSeed, update.NewState)
 
-		t.keyRing.AssertNumberOfCalls(t, "DeriveNextKey", keyCount)
+		require.Eventually(t, func() bool {
+			t.keyRing.AssertNumberOfCalls(
```

Comment:
I don't think this works as expected?
If the `AssertNumberOfCalls` fails, then `t.Fatalf()` is called, which will probably cause the `Eventually()` to not re-try.
So perhaps we can use [`EventuallyWithT`](https://pkg.go.dev/github.com/stretchr/testify/require#EventuallyWithT)?

Reply:
Good catch, I think you're right.

`collect *assert.CollectT` doesn't work with the `t` required by `AssertNumberOfCalls`. Some method is missing. I've stuck with `Eventually` and avoided the `AssertNumberOfCalls` call.

Let me know what you think.